Remotion/Web/Development.WebTesting/HostingStrategies/DockerHostingStrategy.cs

```diff
@@ -32,19 +32,45 @@ namespace Remotion.Web.Development.WebTesting.HostingStrategies
   /// </summary>
   public class DockerHostingStrategy : IHostingStrategy
   {
-    private readonly IisDockerContainerWrapper _iisDockerContainerWrapper;
-
-    public DockerHostingStrategy (
-        [NotNull] ITestSiteLayoutConfiguration testSiteLayoutConfiguration,
-        int port,
-        [NotNull] string dockerImageName,
-        [CanBeNull] string? dockerIsolationMode,
-        TimeSpan dockerPullTimeout,
-        [CanBeNull] string? hostname)
+    private readonly DockerContainerWrapperBase _dockerContainerWrapper;
+
+    public DockerHostingStrategy (DockerContainerWrapperBase dockerContainerWrapper)
     {
-      ArgumentUtility.CheckNotNull("testSiteLayoutConfiguration", testSiteLayoutConfiguration);
-      ArgumentUtility.CheckNotNullOrEmpty("dockerImageName", dockerImageName);
-      ArgumentUtility.CheckNotEmpty("hostname", hostname);
+      ArgumentUtility.CheckNotNull("dockerContainerWrapper", dockerContainerWrapper);
+
+      _dockerContainerWrapper = dockerContainerWrapper;
+    }
+
+    /// <summary>
+    /// Constructor required for direct usage in <see cref="WebTestConfigurationSection"/>.
+    /// </summary>
+    /// <param name="testSiteLayoutConfiguration">The configuration of the used test site.</param>
+    /// <param name="properties">The configuration properties.</param>
+    [UsedImplicitly]
+    public DockerHostingStrategy ([NotNull] ITestSiteLayoutConfiguration testSiteLayoutConfiguration, [NotNull] IReadOnlyDictionary<string, string> properties)
+        : this(CreateContainerWrapper(testSiteLayoutConfiguration, properties))
+    {
+      // TODO RM-8113: Guard used properties against null values.
+    }
+
+    private static DockerContainerWrapperBase CreateContainerWrapper (
+        ITestSiteLayoutConfiguration testSiteLayoutConfiguration,
+        [NotNull] IReadOnlyDictionary<string, string> properties)
+    {
+      ArgumentUtility.CheckNotNull(nameof(testSiteLayoutConfiguration), testSiteLayoutConfiguration);
+      ArgumentUtility.CheckNotNull(nameof(properties), properties);
+
+      var port = int.Parse(properties["port"]);
+      var dockerImageName = properties["dockerImageName"];
+      var dockerIsolationMode = properties["dockerIsolationMode"];
+      var dockerPullTimeout = TimeSpan.Parse(properties["dockerPullTimeout"]);
+      var hostname = properties["hostname"];
+#if NET6_0_OR_GREATER
+      var innerType = properties.GetValueOrDefault("innerType");
+#else
+      string? innerType;
+      properties.TryGetValue("innerType", out innerType);
+#endif
```

Comment:
is this conditional helpful since it's just syntax?

Reply:
Since Netframework is dropping directly after this, I think it makes sense to go for the new syntax already.

Reply:
okay